common/console.c

```diff
@@ -122,31 +122,39 @@ void register_console_callback(console_callback_t cb, void *arg) {
     console_callbacks[num_console_callbacks++].arg = arg;
 }
 
-void __noreturn panic(const char *fmt, ...) {
-    va_list args;
+static void oops_print(const char *fmt, va_list args, const char *type) {
+    static const char stars[] = "***********************************";
+    size_t slen = sizeof(stars) - 1;
+    size_t tlen = strlen(type);
+    int s1_len, s2_len;
+
+    if (tlen > slen - 4)
+        tlen = slen - 4;
```

Comment:
probably impossible, unless we're introducing a some "oops-holy-shit-wtf-bbq-help" type, but I guess nice to to handle.  in fact, even _that_ wouldn't be long enough for this to be useful hehe.. I guess it's not possible to fail at build-time if a too long `type` is used...

Reply:
Yes, sure, won't be required for the current users. But I tend to write code that handles all cases just fine, that's why I went with this to make sure even overly long `type` strings will, at least, be surrounded by a star including whitespace.